Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package de.rki.coronawarnapp.ui.qrcode.fullscreen
+
+import android.graphics.Bitmap
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.presencetracing.checkins.qrcode.QrCodeGenerator
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import timber.log.Timber
+
+class QrCodeFullScreenViewModel @AssistedInject constructor(
+    @Assisted private val qrcodeText: String,
+    private val qrCodeGenerator: QrCodeGenerator,
+    dispatcherProvider: DispatcherProvider
+) : CWAViewModel(dispatcherProvider) {
+
+    private val qrCodeBitmap = MutableLiveData<Bitmap>()
+    val qrcode: LiveData<Bitmap> = qrCodeBitmap
+    val immersiveMode = SingleLiveEvent<Boolean>()
+
+    init {
+        generateQrCode()
+    }
+
+    private fun generateQrCode() = launch {
+        try {
+            qrCodeBitmap.postValue(qrCodeGenerator.createQrCode(qrcodeText))
+        } catch (e: Exception) {
+            Timber.d(e, "generateQrCode failed")
+            e.report(ExceptionCategory.UI)
```

Comment:
The `e.report` is really just a not so nice workaround that we need to use less.
If we stay on this screen, in the error case, can we just do a simple `errorEvents: SingleLiveEvent` on this screen? Show the error where it happens?

Reply:
Removed , No need in this case